apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Keys.cs

```diff
@@ -275,49 +302,72 @@ private V1.KeyRange ToProtobuf()
     /// </summary>
     public sealed class Key
     {
-        /// <summary>
-        /// The values of this key.
-        /// </summary>
-        internal ListValue KeyParts { get; }
+        internal SpannerParameterCollection Parameters { get; }
 
         /// <summary>
         /// Creates a key from the values in the parameter collection.
         /// </summary>
         /// <param name="keyParts">The values that define the key</param>
         internal Key(SpannerParameterCollection keyParts)
         {
-            KeyParts = ToListValue(keyParts);
+            Parameters = keyParts.Clone();
```

Comment:
Cloning here does not guarantee inmutability, the list of parameters is itself created new but the parameters are shallowed copied.

I prefer if we don't stand in middle ground, so let's not clone anything here.
@jskeet thoughts?